ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -76,7 +77,14 @@ object ParallelIndexerFactory {
             // The life-cycle of such connections matches the life-cycle of a protectedExecution
             dataSource = dataSourceStorageBackend.createDataSource(dbConfig.dataSourceConfig)
           } yield HaCoordinator.databaseLockBasedHaCoordinator(
-            connectionFactory = () => dataSource.getConnection,
+            connectionFactory = () =>
+              dataSource.getConnection
+                .tap(
```

Comment:
What about rewriting it as
```
            connectionFactory = () => {
              val connection = dataSource.getConnection
              connection.setNetworkTimeout(DirectExecutor, haConfig.mainLockCheckerJdbcNetworkTimeoutMillis)
              connection
            },
```
to make it easier to grasp when reading the code?


```
object DirectExecutor extends Executor {
  override def execute(command: Runnable): Unit = command.run()
}
```

Reply:
sure